front-end/src/telemetry/notification.jsx

```diff
@@ -27,7 +27,8 @@ export default class NotificationSettings extends React.Component {
   updateTo () {
     return function (ev) {
       const config = this.state.config
```

Comment:
this config is just a pointer to state.config, so the subsequent lines are directly changing a state variable.  We may need to clone the config object in order to call setState (just a theory).